src/Namecheap/Helper/NamecheapApi.php

```diff
@@ -0,0 +1,602 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Upmind\ProvisionProviders\DomainNames\Namecheap\Helper;
+
+use Carbon\Carbon;
+use GuzzleHttp\Client;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Http\Request;
+use InvalidArgumentException;
+use SimpleXMLElement;
+use Upmind\ProvisionBase\Exception\ProvisionFunctionError;
+use Upmind\ProvisionProviders\DomainNames\Data\ContactData;
+use Upmind\ProvisionProviders\DomainNames\Data\ContactParams;
+use Upmind\ProvisionProviders\DomainNames\Data\DacDomain;
+use Upmind\ProvisionProviders\DomainNames\Data\NameserversResult;
+use Upmind\ProvisionProviders\DomainNames\Helper\Utils;
+use Upmind\ProvisionProviders\DomainNames\Namecheap\Data\NamecheapConfiguration;
+
+/**
+ * Class NamecheapCommand
+ *
+ * @package Upmind\ProvisionProviders\DomainNames\Namecheap\Helper
+ */
+class NamecheapApi
+{
+    /**
+     * Allowed contact types
+     */
+    protected const ALLOWED_CONTACT_TYPES = ['registrant', 'tech', 'admin', 'auxbilling'];
+
+    public const NO_LOCK_TLDS = [
+        'io',
+    ];
+
+    public const TRANSFER_TLD = [
+        "biz", "ca", "cc", "co",
+        "co.uk", "com", "com.es", "com.pe",
+        "es", "in", "info", "me",
+        "me.uk", "mobi", "net", "net.pe",
+        "nom.es", "org", "org.es", "org.pe",
+        "org.uk", "pe", "tv", "us",
+    ];
+
+    /**
+     * Contact Types
+     */
+    public const CONTACT_TYPE_REGISTRANT = 'Registrant';
+    public const CONTACT_TYPE_TECH = 'Tech';
+    public const CONTACT_TYPE_ADMIN = 'Admin';
+    public const CONTACT_TYPE_BILLING = 'AuxBilling';
+
+    protected Client $client;
+
+    protected NamecheapConfiguration $configuration;
+
+    public function __construct(Client $client, NamecheapConfiguration $configuration)
+    {
+        $this->client = $client;
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * @param  string  $domainList
+     *
+     * @return array
+     */
+    public function checkMultipleDomains(string $domainList): array
+    {
+        $params = [
+            'command'    => 'namecheap.domains.check',
+            'DomainList' => $domainList,
+        ];
+
+        $response = $this->makeRequest($params);
+
+        $dacDomains = [];
+
+        foreach ($response->children() as $domain) {
+            $domainName = (string) $domain->attributes()->Domain;
+
+            $available = (string) $domain->attributes()->Available === "true";
+
+            $canTransfer = false;
+
+            if (!$available) {
+                $canTransfer = !$this->getRegistrarLockStatus($domainName);
```

Comment:
This will fail when the domain name exists outside of the Namecheap account. E.g., try google.com

Instead of this, just assume the domain can be transferred if it is not available to register i.e., `'can_transfer' => !$available,`